fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/MetricUnionAvailability.java

```diff
@@ -81,28 +75,15 @@
      *
      * @param availabilities  A set of <tt>Availabilities</tt> whose Dimension schemas are (typically) the same and
      * the Metric columns are unique(i.e. no overlap) on every availability
-     * @param columns  The set of all configured columns, including dimension columns, that metric union availability
+     * @param availabilitiesToMetricNames  A map of all availabilities to set of metric names
      * will respond with
      */
-    public MetricUnionAvailability(@NotNull Set<Availability> availabilities, @NotNull Set<Column> columns) {
+    public MetricUnionAvailability(@NotNull Set<Availability> availabilities,
+                                   @NotNull Map<Availability, Set<String>> availabilitiesToMetricNames) {
         super(availabilities.stream());
-        metricNames = Utils.getSubsetByType(columns, MetricColumn.class).stream()
-                .map(MetricColumn::getName)
-                .collect(Collectors.toSet());
+        metricNames = availabilitiesToMetricNames.values().stream().flatMap(Set::stream).collect(Collectors.toSet());
 
-        // Construct a map of availability to its assigned metric
-        // by intersecting its underlying datasource metrics with table configured metrics
-        availabilitiesToMetricNames = availabilities.stream()
-                .collect(
-                        Collectors.toMap(
-                                Function.identity(),
-                                availability ->
-                                        Sets.intersection(
-                                                availability.getAllAvailableIntervals().keySet(),
-                                                metricNames
-                                        )
-                        )
-                );
+        this.availabilitiesToMetricNames = availabilitiesToMetricNames;
```

Comment:
`availabilitiesToMetricNames ` shall be wrapped into an `immutableMap`.

Reply:
wrapping it while constructing it in 'MetricUnionCompositeTableDefinition'